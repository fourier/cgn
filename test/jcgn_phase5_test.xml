<?xml version="1.0" encoding="UTF-8"?>
<!-- default namespace for elements: cgn -->
<root xmlns="https://github.com/fourier/cgn" 
      xmlns:cgn="https://github.com/fourier/cgn"
      xmlns:jcgn="https://github.com/fourier/cgn/java">

  <!-- this test tests the 5th transformation phase -->

  <!-- in 5th transformation phase we attach the temporary
       child jtmp:object-fields to every object.
       So here we test if jtmp:object-fields constructed properly
  -->
  <!-- test if nothing happens with the normal field types,
       like primitive types or arrays of primitive types
  -->
  <objects cgn:package="com.test.java.phase5.field.types.normal">
    <object cgn:name="Test1">
      <field cgn:name="field1" cgn:type="int"/>
      <field cgn:name="field2" cgn:type="double"/>
      <field cgn:name="field3" cgn:type="[int]"/>
      <field cgn:name="field4" cgn:type="[string]"/>
      <field cgn:name="field5" />
      <field cgn:name="field6" cgn:type="date"/>
    </object>
  </objects>

  <!-- test field types which are not arrays -->
  <objects cgn:package="com.test.java.phase5.fqdn.not.arrays">
    <object cgn:name="SomeObject"/>
    <object cgn:name="SomeObject1">
      <!-- test with full FQDN -->
      <field cgn:name="object-someobject" cgn:type="com.test.java.phase5.fqdn.one.SomeObject1"/>
      <!-- test unknown type with FQDN -->
      <field cgn:name="object-unknown-fqdn-type" cgn:type="com.unknown.Type"/>
      <!-- test unknown type without FQDN -->
      <field cgn:name="object-unknown-type" cgn:type="UnknownType"/>
      <!-- test resolving unambiguous FQDN -->
      <field cgn:name="object-someobject2" cgn:type="SomeObject2"/>
      <!-- test resolving ambiguous FQDN where object is in the same package -->
      <field cgn:name="object-ambigious-same-package" cgn:type="SomeObject"/>
      <!-- test resolving ambiguous FQDN where object is not in the same package -->
      <field cgn:name="object-ambigious-another-package1" cgn:type="SomeObject3"/>
      <field cgn:name="object-ambigious-another-package2" cgn:type="com.test.java.phase5.fqdn.package2.SomeObject3"/>
    </object>

  </objects>

  <!-- test field types which are arrays -->
  <objects cgn:package="com.test.java.phase5.fqdn.arrays">
    <object cgn:name="SomeObject"/>
    <object cgn:name="SomeObject1">
      <!-- test with full FQDN -->
      <field cgn:name="object-someobject" cgn:type="[com.test.java.phase5.fqdn.one.SomeObject1]"/>
      <!-- test unknown type with FQDN -->
      <field cgn:name="object-unknown-fqdn-type" cgn:type="[com.unknown.Type]"/>
      <!-- test unknown type without FQDN -->
      <field cgn:name="object-unknown-type" cgn:type="[UnknownType]"/>
      <!-- test resolving unambiguous FQDN -->
      <field cgn:name="object-someobject2" cgn:type="[SomeObject2]"/>
      <!-- test resolving ambiguous FQDN where object is in the same package -->
      <field cgn:name="object-ambigious-same-package" cgn:type="[SomeObject]"/>
      <!-- test resolving ambiguous FQDN where object is not in the same package -->
      <field cgn:name="object-ambigious-another-package1" cgn:type="[SomeObject3]"/>
      <field cgn:name="object-ambigious-another-package2" cgn:type="[com.test.java.phase5.fqdn.package2.SomeObject3]"/>

    </object>

  </objects>

  

  <!-- help objects -->
  <objects>
    <object cgn:name="SomeObject1" cgn:package="com.test.java.phase5.fqdn.one"/>
    <object cgn:name="SomeObject2" cgn:package="com.test.java.phase5.fqdn.two"/>
    <object cgn:name="SomeObject3" cgn:package="com.test.java.phase5.fqdn.package1"/>
    <object cgn:name="SomeObject3" cgn:package="com.test.java.phase5.fqdn.package2"/>
  </objects>

  
</root>

