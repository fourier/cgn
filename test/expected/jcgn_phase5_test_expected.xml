<?xml version="1.0" encoding="UTF-8"?>
<!-- default namespace for elements: cgn -->
<root xmlns="https://github.com/fourier/cgn"
      xmlns:cgn="https://github.com/fourier/cgn"
      xmlns:jcgn="https://github.com/fourier/cgn/java"><!-- this test tests the 5th transformation phase --><!-- in 5th transformation phase we attach the temporary
       child jtmp:object-fields to every object.
       So here we test if jtmp:object-fields constructed properly
  --><!-- test if nothing happens with the normal field types,
       like primitive types or arrays of primitive types
  -->
   <objects cgn:package="com.test.java.phase5.field.types.normal"
            cgn:copyright=""
            cgn:author=""
            cgn:read-only="false"
            cgn:json="false"
            cgn:is-set="false"
            jcgn:builder="false"
            jcgn:parcelable="false"
            jcgn:date-type="java">
      <object cgn:name="Test1"
              cgn:copyright=""
              cgn:author=""
              cgn:package="com.test.java.phase5.field.types.normal"
              cgn:read-only="false"
              cgn:json="false"
              cgn:is-set="false"
              jcgn:builder="false"
              jcgn:parcelable="false"
              jcgn:date-type="java">
         <field cgn:name="field1" cgn:type="int"/>
         <field cgn:name="field2" cgn:type="double"/>
         <field cgn:name="field3" cgn:type="[int]"/>
         <field cgn:name="field4" cgn:type="[string]"/>
         <field cgn:name="field5" cgn:type="string"/>
         <field cgn:name="field6" cgn:type="date" jcgn:date-type="java"/>
         <field cgn:name="field7" cgn:type="[date]" jcgn:date-type="java"/>
         <field cgn:name="field8" cgn:type="date" jcgn:date-type="joda"/>
         <field cgn:name="field9" cgn:type="[date]" jcgn:date-type="joda"/>
         <jtmp:object-fields xmlns:jtmp="https://github.com/fourier/cgn/java/temp">
            <field xmlns=""
                   name="field1"
                   type="int"
                   java-name="mField1"
                   java-type="int"
                   java-full-type="int"/>
            <field xmlns=""
                   name="field2"
                   type="double"
                   java-name="mField2"
                   java-type="double"
                   java-full-type="double"/>
            <field xmlns=""
                   name="field3"
                   type="[int]"
                   java-name="mField3"
                   java-type="java.util.ArrayList&lt;Integer&gt;"
                   java-full-type="java.util.ArrayList&lt;Integer&gt;"/>
            <field xmlns=""
                   name="field4"
                   type="[string]"
                   java-name="mField4"
                   java-type="java.util.ArrayList&lt;String&gt;"
                   java-full-type="java.util.ArrayList&lt;String&gt;"/>
            <field xmlns=""
                   name="field5"
                   type="string"
                   java-name="mField5"
                   java-type="String"
                   java-full-type="String"/>
            <field xmlns=""
                   name="field6"
                   type="date"
                   java-name="mField6"
                   java-type="java.util.Date"
                   java-full-type="java.util.Date"/>
            <field xmlns=""
                   name="field7"
                   type="[date]"
                   java-name="mField7"
                   java-type="java.util.ArrayList&lt;java.util.Date&gt;"
                   java-full-type="java.util.ArrayList&lt;java.util.Date&gt;"/>
            <field xmlns=""
                   name="field8"
                   type="date"
                   java-name="mField8"
                   java-type="org.joda.time.DateTime"
                   java-full-type="org.joda.time.DateTime"/>
            <field xmlns=""
                   name="field9"
                   type="[date]"
                   java-name="mField9"
                   java-type="java.util.ArrayList&lt;org.joda.time.DateTime&gt;"
                   java-full-type="java.util.ArrayList&lt;org.joda.time.DateTime&gt;"/>
         </jtmp:object-fields>
      </object>
   </objects>
   <!-- test field types which are not arrays -->
   <objects cgn:package="com.test.java.phase5.fqdn.not.arrays"
            cgn:copyright=""
            cgn:author=""
            cgn:read-only="false"
            cgn:json="false"
            cgn:is-set="false"
            jcgn:builder="false"
            jcgn:parcelable="false"
            jcgn:date-type="java">
      <object cgn:name="SomeObject"
              cgn:copyright=""
              cgn:author=""
              cgn:package="com.test.java.phase5.fqdn.not.arrays"
              cgn:read-only="false"
              cgn:json="false"
              cgn:is-set="false"
              jcgn:builder="false"
              jcgn:parcelable="false"
              jcgn:date-type="java">
         <jtmp:object-fields xmlns:jtmp="https://github.com/fourier/cgn/java/temp"/>
      </object>
      <object cgn:name="SomeObject1"
              cgn:copyright=""
              cgn:author=""
              cgn:package="com.test.java.phase5.fqdn.not.arrays"
              cgn:read-only="false"
              cgn:json="false"
              cgn:is-set="false"
              jcgn:builder="false"
              jcgn:parcelable="false"
              jcgn:date-type="java"><!-- test with full FQDN -->
         <field cgn:name="object-someobject"
                cgn:type="com.test.java.phase5.fqdn.one.SomeObject1"/>
         <!-- test unknown type with FQDN -->
         <field cgn:name="object-unknown-fqdn-type" cgn:type="com.unknown.Type"/>
         <!-- test unknown type without FQDN -->
         <field cgn:name="object-unknown-type" cgn:type="UnknownType"/>
         <!-- test resolving unambiguous FQDN -->
         <field cgn:name="object-someobject2"
                cgn:type="com.test.java.phase5.fqdn.two.SomeObject2"/>
         <!-- test resolving ambiguous FQDN where object is in the same package -->
         <field cgn:name="object-ambigious-same-package"
                cgn:type="com.test.java.phase5.fqdn.not.arrays.SomeObject"/>
         <!-- test resolving ambiguous FQDN where object is not in the same package -->
         <field cgn:name="object-ambigious-another-package1"
                cgn:type="com.test.java.phase5.fqdn.package1.SomeObject3"/>
         <field cgn:name="object-ambigious-another-package2"
                cgn:type="com.test.java.phase5.fqdn.package2.SomeObject3"/>
         <jtmp:object-fields xmlns:jtmp="https://github.com/fourier/cgn/java/temp">
            <field xmlns=""
                   name="object-someobject"
                   type="com.test.java.phase5.fqdn.one.SomeObject1"
                   java-name="mObjectSomeobject"
                   java-type="SomeObject1"
                   java-full-type="com.test.java.phase5.fqdn.one.SomeObject1"/>
            <field xmlns=""
                   name="object-unknown-fqdn-type"
                   type="com.unknown.Type"
                   java-name="mObjectUnknownFqdnType"
                   java-type="Type"
                   java-full-type="com.unknown.Type"/>
            <field xmlns=""
                   name="object-unknown-type"
                   type="UnknownType"
                   java-name="mObjectUnknownType"
                   java-type="UnknownType"
                   java-full-type="UnknownType"/>
            <field xmlns=""
                   name="object-someobject2"
                   type="com.test.java.phase5.fqdn.two.SomeObject2"
                   java-name="mObjectSomeobject2"
                   java-type="SomeObject2"
                   java-full-type="com.test.java.phase5.fqdn.two.SomeObject2"/>
            <field xmlns=""
                   name="object-ambigious-same-package"
                   type="com.test.java.phase5.fqdn.not.arrays.SomeObject"
                   java-name="mObjectAmbigiousSamePackage"
                   java-type="SomeObject"
                   java-full-type="com.test.java.phase5.fqdn.not.arrays.SomeObject"/>
            <field xmlns=""
                   name="object-ambigious-another-package1"
                   type="com.test.java.phase5.fqdn.package1.SomeObject3"
                   java-name="mObjectAmbigiousAnotherPackage1"
                   java-type="com.test.java.phase5.fqdn.package1.SomeObject3"
                   java-full-type="com.test.java.phase5.fqdn.package1.SomeObject3"/>
            <field xmlns=""
                   name="object-ambigious-another-package2"
                   type="com.test.java.phase5.fqdn.package2.SomeObject3"
                   java-name="mObjectAmbigiousAnotherPackage2"
                   java-type="com.test.java.phase5.fqdn.package2.SomeObject3"
                   java-full-type="com.test.java.phase5.fqdn.package2.SomeObject3"/>
         </jtmp:object-fields>
      </object>
   </objects>
   <!-- test field types which are arrays -->
   <objects cgn:package="com.test.java.phase5.fqdn.arrays"
            cgn:copyright=""
            cgn:author=""
            cgn:read-only="false"
            cgn:json="false"
            cgn:is-set="false"
            jcgn:builder="false"
            jcgn:parcelable="false"
            jcgn:date-type="java">
      <object cgn:name="SomeObject"
              cgn:copyright=""
              cgn:author=""
              cgn:package="com.test.java.phase5.fqdn.arrays"
              cgn:read-only="false"
              cgn:json="false"
              cgn:is-set="false"
              jcgn:builder="false"
              jcgn:parcelable="false"
              jcgn:date-type="java">
         <jtmp:object-fields xmlns:jtmp="https://github.com/fourier/cgn/java/temp"/>
      </object>
      <object cgn:name="SomeObject1"
              cgn:copyright=""
              cgn:author=""
              cgn:package="com.test.java.phase5.fqdn.arrays"
              cgn:read-only="false"
              cgn:json="false"
              cgn:is-set="false"
              jcgn:builder="false"
              jcgn:parcelable="false"
              jcgn:date-type="java"><!-- test with full FQDN -->
         <field cgn:name="object-someobject"
                cgn:type="[com.test.java.phase5.fqdn.one.SomeObject1]"/>
         <!-- test unknown type with FQDN -->
         <field cgn:name="object-unknown-fqdn-type" cgn:type="[com.unknown.Type]"/>
         <!-- test unknown type without FQDN -->
         <field cgn:name="object-unknown-type" cgn:type="[UnknownType]"/>
         <!-- test resolving unambiguous FQDN -->
         <field cgn:name="object-someobject2"
                cgn:type="[com.test.java.phase5.fqdn.two.SomeObject2]"/>
         <!-- test resolving ambiguous FQDN where object is in the same package -->
         <field cgn:name="object-ambigious-same-package"
                cgn:type="[com.test.java.phase5.fqdn.arrays.SomeObject]"/>
         <!-- test resolving ambiguous FQDN where object is not in the same package -->
         <field cgn:name="object-ambigious-another-package1"
                cgn:type="[com.test.java.phase5.fqdn.package1.SomeObject3]"/>
         <field cgn:name="object-ambigious-another-package2"
                cgn:type="[com.test.java.phase5.fqdn.package2.SomeObject3]"/>
         <jtmp:object-fields xmlns:jtmp="https://github.com/fourier/cgn/java/temp">
            <field xmlns=""
                   name="object-someobject"
                   type="[com.test.java.phase5.fqdn.one.SomeObject1]"
                   java-name="mObjectSomeobject"
                   java-type="java.util.ArrayList&lt;SomeObject1&gt;"
                   java-full-type="java.util.ArrayList&lt;com.test.java.phase5.fqdn.one.SomeObject1&gt;"/>
            <field xmlns=""
                   name="object-unknown-fqdn-type"
                   type="[com.unknown.Type]"
                   java-name="mObjectUnknownFqdnType"
                   java-type="java.util.ArrayList&lt;Type&gt;"
                   java-full-type="java.util.ArrayList&lt;com.unknown.Type&gt;"/>
            <field xmlns=""
                   name="object-unknown-type"
                   type="[UnknownType]"
                   java-name="mObjectUnknownType"
                   java-type="java.util.ArrayList&lt;UnknownType&gt;"
                   java-full-type="java.util.ArrayList&lt;UnknownType&gt;"/>
            <field xmlns=""
                   name="object-someobject2"
                   type="[com.test.java.phase5.fqdn.two.SomeObject2]"
                   java-name="mObjectSomeobject2"
                   java-type="java.util.ArrayList&lt;SomeObject2&gt;"
                   java-full-type="java.util.ArrayList&lt;com.test.java.phase5.fqdn.two.SomeObject2&gt;"/>
            <field xmlns=""
                   name="object-ambigious-same-package"
                   type="[com.test.java.phase5.fqdn.arrays.SomeObject]"
                   java-name="mObjectAmbigiousSamePackage"
                   java-type="java.util.ArrayList&lt;SomeObject&gt;"
                   java-full-type="java.util.ArrayList&lt;com.test.java.phase5.fqdn.arrays.SomeObject&gt;"/>
            <field xmlns=""
                   name="object-ambigious-another-package1"
                   type="[com.test.java.phase5.fqdn.package1.SomeObject3]"
                   java-name="mObjectAmbigiousAnotherPackage1"
                   java-type="java.util.ArrayList&lt;com.test.java.phase5.fqdn.package1.SomeObject3&gt;"
                   java-full-type="java.util.ArrayList&lt;com.test.java.phase5.fqdn.package1.SomeObject3&gt;"/>
            <field xmlns=""
                   name="object-ambigious-another-package2"
                   type="[com.test.java.phase5.fqdn.package2.SomeObject3]"
                   java-name="mObjectAmbigiousAnotherPackage2"
                   java-type="java.util.ArrayList&lt;com.test.java.phase5.fqdn.package2.SomeObject3&gt;"
                   java-full-type="java.util.ArrayList&lt;com.test.java.phase5.fqdn.package2.SomeObject3&gt;"/>
         </jtmp:object-fields>
      </object>
   </objects>
   <!-- help objects -->
   <objects cgn:copyright=""
            cgn:author=""
            cgn:package="com.example.cgn"
            cgn:read-only="false"
            cgn:json="false"
            cgn:is-set="false"
            jcgn:builder="false"
            jcgn:parcelable="false"
            jcgn:date-type="java">
      <object cgn:name="SomeObject1"
              cgn:package="com.test.java.phase5.fqdn.one"
              cgn:copyright=""
              cgn:author=""
              cgn:read-only="false"
              cgn:json="false"
              cgn:is-set="false"
              jcgn:builder="false"
              jcgn:parcelable="false"
              jcgn:date-type="java">
         <jtmp:object-fields xmlns:jtmp="https://github.com/fourier/cgn/java/temp"/>
      </object>
      <object cgn:name="SomeObject2"
              cgn:package="com.test.java.phase5.fqdn.two"
              cgn:copyright=""
              cgn:author=""
              cgn:read-only="false"
              cgn:json="false"
              cgn:is-set="false"
              jcgn:builder="false"
              jcgn:parcelable="false"
              jcgn:date-type="java">
         <jtmp:object-fields xmlns:jtmp="https://github.com/fourier/cgn/java/temp"/>
      </object>
      <object cgn:name="SomeObject3"
              cgn:package="com.test.java.phase5.fqdn.package1"
              cgn:copyright=""
              cgn:author=""
              cgn:read-only="false"
              cgn:json="false"
              cgn:is-set="false"
              jcgn:builder="false"
              jcgn:parcelable="false"
              jcgn:date-type="java">
         <jtmp:object-fields xmlns:jtmp="https://github.com/fourier/cgn/java/temp"/>
      </object>
      <object cgn:name="SomeObject3"
              cgn:package="com.test.java.phase5.fqdn.package2"
              cgn:copyright=""
              cgn:author=""
              cgn:read-only="false"
              cgn:json="false"
              cgn:is-set="false"
              jcgn:builder="false"
              jcgn:parcelable="false"
              jcgn:date-type="java">
         <jtmp:object-fields xmlns:jtmp="https://github.com/fourier/cgn/java/temp"/>
      </object>
   </objects>
</root>
